### PR TITLE
Add ReactRelayContext to Usage docs

### PR DESCRIPTION
**Pull request by gfrancischelli**
Took me a few minutes going through error messages and finding [`useContext(ReactRelayContext)`](https://github.com/relay-tools/react-relay-mutation/blob/c08ffce7d256b02612b7df3d9a5de96c136365d4/src/index.tsx#L45) in the source code to realize what was missing in my setup.

It took longer than it could be since I've failed to find any mention to`ReactRelayContext` in [relay.dev](http://relay.dev). Might tinker around relay source to provide a better explanation here 🤔

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,23 @@ function MyComponentWithComponent({ myValue }) {
...
+```js
+import { ReactRelayContext } from "react-relay";
+import environment from "./environment";
+
+function App() {
+  const contextValue = { environment, variables: {} }
+  return (
+    <ReactRelayContext.Provider value={contextValue}/>
+      <MyComponentWithHook/>
+      <MyComponentWithComponent/>
```
Comment (itajaja) [inline README.md]: `MyComponentWithComponent`? 🤔

Comment (gfrancischelli) [reply README.md]: Used the name of the components in the example but it's a bit weird here.

Renamed to  `ComponentWithMutation` for clarity.

File: README.md [hunk 2]
```diff
@@ -26,6 +27,7 @@ function MyComponentWithHook({ myValue }) {
         window.alert(`received ${myMutation.value}`);
       },
     },
+    environment // Not required when provided with ReactRelayContext
```
Comment (gfrancischelli) [inline README.md]: > good point. we actually support passing the environment directly: [/src/index.tsx@`master`#L34](https://github.com/relay-tools/react-relay-mutation/blob/master/src/index.tsx#L34) [](/relay-tools/react-relay-mutation/blob/HEAD@{2019-06-21T18:42:56Z}/src/index.tsx#L34) so maybe we should specify that as well

Now addressing addresses this 👆

**Comment by itajaja** [comment]
you should run prettier locally to fix the CI errors

**Comment by gfrancischelli** [comment]
@itajaja Thanks, I wasn't understanding why CI failed

**Comment by itajaja** [comment]
thanks!

**Comment by taion** [comment]
rewrote this a little to improve clarity

**Pull request merged**